<launch>

  <!-- Launch file for making a turtle move in rviz using turtlebot3_teleop -->

  <!-- Define Default Arguments -->
  <arg name="use_rviz" default="True" doc="The use_rviz parameter controls whether Rviz and the joint_state_publisher are launched (default true)"/>
  <arg name="use_teleop" default="True" doc="The use_teleop parameter controls whether turtlebot3_teleop_keyboard node is launched (default true)"/>

  <!-- Load Turtle Parameters -->
  <rosparam command="load" file="$(find rigid2d)/config/robot_params.yaml" />
  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />
  <rosparam command="load" file="$(find nurtlesim)/config/nurtlesim_params.yaml" />
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find nuturtle_description)/urdf/my_turtlebot3.urdf.xacro" />

  <!-- Load turtlebot3_teleop -->
  <group if="$(arg use_teleop)">
    <include file="$(find turtlebot3_teleop)/launch/turtlebot3_teleop_key.launch" />
  </group>

  <!-- Load odometer_node.cpp Node -->
  <node name="tube_world" pkg="nurtlesim" type="tube_world" output="screen" />

  <!-- Load Robot Model in Rviz -->
  <group if="$(arg use_rviz)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nurtlesim)/config/tube_world.rviz" />
  </group>

</launch>